### PR TITLE
Introduce Network Policies section and add Weave Net

### PR DESCRIPTION
**Pull request by bboreham**
Added a new subdirectory, `network-policies` and included a top-level file with an overview of Network policy and Calico and Weave Net. Separated Calico and Weave-specific parts to new subdirectories.

This PR includes suggestions made in #255 and #265 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: network-policies/readme.adoc [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+test                100% |*************************************************************************************************|   534   0:00:00 ETA
+```
+
+== Teardown
+
+Remove all the resources and the namespace using the command:
+
+  \ # kubectl delete ns ns-1
+  namespace "ns-1" deleted
```
Comment (arun-gupta) [inline network-policies/readme.adoc]: Have you verified that this section would work with both Calico and Weavenet?

Comment (bboreham) [reply network-policies/readme.adoc]: I can run through it with Calico, but ideally we'd get someone more familiar with Calico to vet the changes.

**Comment by arun-gupta** [comment]
@bboreham can you resolve the conflicts?

**Comment by bboreham** [comment]
Rebased off master and bumped Weave Net version to 2.1.3 which I have tested extensively with.

**Comment by arun-gupta** [comment]
I had to merge another critical PR and that threw this one off again. One more try? I promise to merge this one first.

**Comment by bboreham** [comment]
Rebased and squashed down to fewer commits

**Pull request merged**